packages/components/src/lib/controllers/settings-controller.js

```diff
@@ -0,0 +1,264 @@
+// import Platform from '@/lib/utility/platform.js'
+// import { Logger } from 'alpheios-data-models'
+import Options from '@comp/lib/options/options.js'
+
+/**
+ */
+export default class SettingsController {
+  constructor ({ platform } = {}) {
+    if (!platform) {
+      throw new Error('No platform data provided for a settings controller')
+    }
+
+    /**
+     * An object with information about an app environment.
+     *
+     * @type {Platform}
+     */
+    this._platform = platform
+
+    this._storageAdapter = null
+
+    this._featureOptions = null
+    this._resourceOptions = null
+    this._uiOptions = null
+
+    /*
+    A copy of resource options for the lookup UI component
+    This doesn't get reloaded from the storage adapter because
+    we don't expose it to the user via preferences
+     */
+    this._lookupResourceOptions = null
+    this._siteOptions = null
+
+    /**
+     * Holds a shard app-wide API object
+     *
+     * @type {object}
+     * @private
+     */
+    this._api = {}
+
+    // A shared Vuex store object
+    this._store = null
+
+    // An object that stores config settings of a controller
+    this._config = {
+    }
+  }
+
+  async init ({
+    api, store, configServiceUrl, clientId, appName, appVersion, branch, buildNumber, storageAdapter,
+    featureOptionsDefaults, resourceOptionsDefaults, uiOptionsDefaults, siteOptionsDefaults
+  } = {}) {
+    if (!api) {
+      throw new Error('API object is required for a settings controller initialization')
+    }
+    if (!store) {
+      throw new Error('Vuex store is required for a settings controller initialization')
+    }
+
+    this._api = api
+    this._store = store
+    this._storageAdapter = storageAdapter
+
+    this._configServiceUrl = configServiceUrl
+    this._clientId = clientId
+    this._appName = appName
+    this._appVersion = appVersion
+    this._branch = branch
+    this._buildNumber = buildNumber
+
+    this._featureOptionsDefaults = featureOptionsDefaults
+    this._resourceOptionsDefaults = resourceOptionsDefaults
+    this._uiOptionsDefaults = uiOptionsDefaults
+    this._siteOptionsDefaults = siteOptionsDefaults
+
+    const appConfigPromise = this.requestAppOptions()
+    const initOptionsPromise = this.initOptions()
+    /*
+    We do not allow user to configure lookupResourceOptions or siteOptions currently.
+    Because of this, they are initialized separately from the rest of the options.
+    `initOptions()` will be called several times during the app's lifetime,
+    after a user has been logged in, as one example.
+    `initNonConfigurableOptions()` will be called only once,
+    during an initialization of the Settings Controller.
+     */
+    this.initNonConfigurableOptions()
```

Comment:
let's just get rid of lookupResourceOptions and siteOptions since they aren't being used anymore.

Reply:
Agree